src/types.ts

```diff
@@ -0,0 +1,25 @@
+export type Types = "percentage" | "number" | "secs" | "hours";
+
+export interface Metric {
+  id: string;
+  label: string;
+  value: number;
+  type: Types;
+  description: string;
+  category: string;
+  show?: boolean;
+}
+
+export interface ApiData {
+  data: Metric[];
+}
+
+export type SortState = null | "asc" | "desc";
```

Comment:
```suggestion
export type SortState = "asc" | "desc";
```

This is just a _hint_ for you, we can omit this `null` from here and instead, flag the usage with the `?` char, as you did bellow:

```tsx
  sort?: {
    key: string;
    state?: SortState;
  };
```

In that way, you just said that `SortState` can be null, no need to explicit declare that